pkg/pubsub/system.go

```diff
@@ -31,27 +31,42 @@ func (p *System) Publish(ctx context.Context, connection, topic string, msg inte
 			}
 			return c.Publish(context.Background(), msg, topic)
 		}
+		log.Info("Connection is not initialized", "name", connection)
 	} else {
 		log.Info("No pub sub tools are enabled")
 	}
 	return nil
 }
 
-func (p *System) InitiateConnection(ctx context.Context, newConnection client.InitiateConnection, data map[string]string) error {
-	for conn := range data {
-		var config interface{}
-		err := json.Unmarshal([]byte(data[conn]), &config)
-		if err != nil {
-			return err
-		}
-		tmp, err := newConnection(ctx, config)
-		if err != nil {
-			return err
-		}
-		if p.connections == nil {
-			p.connections = map[string]connection.Connection{}
+func (p *System) InitiateConnection(ctx context.Context, newConnection client.InitiateConnection, data map[string]string, name string) error {
+	var config interface{}
+	err := json.Unmarshal([]byte(data["config"]), &config)
```

Comment:
IMO the caller should be the one to unpack "config", this will give flexibility for consumers of this code to package config in different ways.